common-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptService.java

```diff
@@ -69,29 +71,38 @@ public static String modifyMultipleMatchKeyword(String query){
             return null;
         }
         String[] keywords = query.split("[,+\\s+]");
-        for(int i = 0; i < keywords.length; i++){
-            String key = keywords[i];
-            if(key.length() < 3 && !key.isEmpty()){
-                key = "\"" + key + "\"";
-                keywords[i] = key;
-            }
-        }
-
-        StringBuilder query2 = new StringBuilder();
-        for(String key : keywords){
+        List<String> temp = new ArrayList<>();
+        for (String key: keywords) {
             if(!key.isEmpty()){
-                if(query2.length()==0){
-                    query2.append("+");
-                    query2.append(key);
-                }else if(key.contains("\"")){
-                    query2.append(key);
-                }else{
-                    query2.append("+");
-                    query2.append(key);
+                int si = query.indexOf(key);
+                if(si-1 >= 0 && si-1 <= query.length()){
+                    String pre = query.substring(si-1,si);
+                    if(pre.equals("+")) {
+                        temp.add(new String(pre+key));
+                        continue;
+                    }
+                }
+                if(key.contains("-") && !temp.contains(key)){
+                    temp.add(key);
+                }else if(key.contains("*") && key.length() > 1){
```

Comment:
Nit: Some of the spacing here is wacky, actually. Do you mind addressing some of that? (Space after close curly braces and before open curly braces. And probably before and after operators.